--- a/Model/ResourceModel/AbstractResource.php
+++ b/Model/ResourceModel/AbstractResource.php
@@ -502,7 +502,7 @@ abstract class AbstractResource extends \Magento\Eav\Model\Entity\AbstractEntity
                 $staticTable,
                 $staticAttributes
             )->join(
-                ['e' => $this->getTable('catalog_product_entity')],
+                ['e' => $this->getTable($this->getEntityTable())],
                 'e.' . $this->getLinkField() . ' = ' . $staticTable . '.' . $this->getLinkField()
             )->where(
                 'e.entity_id = :entity_id'
@@ -523,7 +523,7 @@ abstract class AbstractResource extends \Magento\Eav\Model\Entity\AbstractEntity
                 $select = $connection->select()
                     ->from(['default_value' => $table], ['attribute_id'])
                     ->join(
-                        ['e' => $this->getTable('catalog_product_entity')],
+                        ['e' => $this->getTable($this->getEntityTable())],
                         'e.' . $this->getLinkField() . ' = ' . 'default_value.' . $this->getLinkField(),
                         ''
                     )->where('default_value.attribute_id IN (?)', array_keys($_attributes))
